api/v1alpha1/reaper_types.go

```diff
@@ -106,6 +115,21 @@ type ReaperSpec struct {
 
 	// Tolerations applied to the Reaper pods
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// InitContainerConfig encapsulates configs applied to the Reaper init container
+	InitContainerConfig InitContainerConfig `json:"initContainerConfig,omitempty" yaml:"initContainerConfig,omitempty"`
+
+	// SecurityContext applied to the Reaper non-init container
+	SecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
+
+	// PodSecurityContext applied to Reaper pods
+	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+
+	// VolumeMounts for the Reaper non-init container
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
```

Comment:
Same as for the init container, we don't need to expose volume mount and volumes. It probably doesn't make a whole lot of sense to do so. We do want a volume and volume mount for the Reaper config file. By default it can be an EmptyDir volume. We don't even need to expose the volume and mount, but if we decide to expose the volume, that would allow the user to provide his own config via a ConfigMap for example.

I just checked and found that the Reaper config file is stored at `/etc/cassandra-reaper.yml`. In order to make the volume mount easier, I think it would be easier to have it at `/etc/reaper/cassandra-reaper.yml`. Then we can mount the volume at `/etc/reaper`. This will require changes in the upstream Reaper image though.

Reply:
👍🏻